src/compiler.ts

```diff
@@ -8629,6 +8629,13 @@ export class Compiler extends DiagnosticEmitter {
       );
       return module.unreachable();
     }
+    if (classReference.kind == ElementKind.INTERFACE) {
+      this.error(
+        DiagnosticCode.Not_implemented_0,
+        expression.range, "Interface hidden classes"
+      );
+      return module.unreachable();
```

Comment:
Let's also set `this.currentType` here even though emitting the error. Otherwise its value is undefined and can lead to unwanted diagnostics down the road. Could just move the next two lines below up.